### PR TITLE
Implement Query Method, Handle Optional Attendees

### PR DESCRIPTION
**Pull request by jessdejong**
I implemented the query method in the FindMeetingQuery java file, and added functionality to handle optional attendees.

I used a PriorityQueue, which simplified things in my mind, but not necessarily in code. I tried to make it more straightforward in comparison to my last pull request. (referring to nik's comment on the other PR, that I deleted)

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/EventTime.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.lang.Comparable;
+
+/** Class that defines nodes for the start and end times of Events */
+public final class EventTime implements Comparable<EventTime> {
```
Comment (estep-google) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/EventTime.java:20]: Is this your writing or something you pulled in?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,124 @@
...
+      EventTime currentEventTime = eventTimes.poll();
+
+      // Check if any ongoing event conflicts with the optional or required attendees
+      boolean meetingConflictsReqAttendees = false;
+      boolean meetingConflictsOptAttendees = false;
+      for (Event event : ongoingEvents) {
+        for (String attendee : event.getAttendees()) {
+          if (requiredAttendees.contains(attendee)) {
+            meetingConflictsReqAttendees = true;
+            meetingConflictsOptAttendees = true;
```
Comment (estep-google) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:64]: Conceptually, I don't like this line. `requiredAttendees` are not optional, so why are we marking a conflict for optional attendees?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -121,6 +123,64 @@ public void everyAttendeeIsConsidered() {
     Assert.assertEquals(expected, actual);
   }
 
+  @Test
+  public void everyAttendeeAndOptionalAttendeeIsConsidered() {
+    // Have each required attendee have different events. An optional attendee has a whole-day event.
```
Comment (estep-google) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:128]: I really like these diagrams you've added here. This makes understanding the test case crystal clear.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/EventTime.java (hunk 1 above)
Comment (jessdejong) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/EventTime.java:20]: I wrote that class, is that what you're asking?

**Comment by jessdejong** [comment]
Added one change that addressed your request!

**Pull request merged**